### PR TITLE
[EventEngine] Implement Windows Native Resolver

### PR DESCRIPTION
**Pull request by drfloob**

### CODE REVIEW

File: src/core/lib/event_engine/windows/native_windows_dns_resolver.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#ifndef GRPC_SRC_CORE_LIB_EVENT_ENGINE_WINDOWS_NATIVE_WINDOWS_DNS_RESOLVER_H
+#define GRPC_SRC_CORE_LIB_EVENT_ENGINE_WINDOWS_NATIVE_WINDOWS_DNS_RESOLVER_H
+
+#include <grpc/support/port_platform.h>
+
+#ifdef GPR_WINDOWS
+
+#include <grpc/event_engine/event_engine.h>
+
+#include "src/core/lib/event_engine/ref_counted_dns_resolver_interface.h"
```
Comment (yijiem) [inline src/core/lib/event_engine/windows/native_windows_dns_resolver.h:23]: This is probably not needed now.

File: src/core/BUILD [hunk 2]
```diff
@@ -2222,6 +2222,21 @@ grpc_cc_library(
...
+)
+
+grpc_cc_library(
+    name = "windows_native_resolver",
+    srcs = ["lib/event_engine/windows/native_windows_dns_resolver.cc"],
+    hdrs = ["lib/event_engine/windows/native_windows_dns_resolver.h"],
+    external_deps = ["absl/strings:str_format"],
+    deps = [
+        "error",
+        "ref_counted_dns_resolver_interface",
```
Comment (yijiem) [inline src/core/BUILD:2238]: Not needed.

**Review by yijiem — APPROVED** [review]
Thanks AJ!

File: src/core/lib/event_engine/windows/native_windows_dns_resolver.h (hunk 1 above)
Comment (drfloob) [reply src/core/lib/event_engine/windows/native_windows_dns_resolver.h]: Good catch, removed.

File: src/core/BUILD (hunk 2 above)
Comment (drfloob) [reply src/core/BUILD]: Good catch, removed.

**Review by yijiem — APPROVED** [review]

**Pull request closed without merging**